tests/bluetooth/controller/ctrl_cte_req/src/main.c

```diff
@@ -616,6 +616,751 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
 }
 
+/* Tests related with PHY update procedure and CTE request procedure "collision" */
+
+#define PREFER_S2_CODING 0U
+#define HOST_INITIATED 1U
+#define PHY_UPDATE_INSTANT_DELTA 6
+
+#define CHECK_PREF_PHY_STATE(_conn, _tx, _rx)                                                      \
+	do {                                                                                       \
+		zassert_equal(_conn.phy_pref_rx, _rx,                                              \
+			      "Preferred RX PHY mismatch %d (actual) != %d (expected)",            \
+			      _conn.phy_pref_rx, _rx);                                             \
+		zassert_equal(_conn.phy_pref_tx, _tx,                                              \
+			      "Preferred TX PHY mismatch %d (actual) != %d (expected)",            \
+			      _conn.phy_pref_tx, _tx);                                             \
+	} while (0)
+
+#define CHECK_CURRENT_PHY_STATE(_conn, _tx, _flags, _rx)                                           \
+	do {                                                                                       \
+		zassert_equal(_conn.lll.phy_rx, _rx,                                               \
+			      "Current RX PHY mismatch %d (actual) != %d (expected)",              \
+			      _conn.lll.phy_rx, _rx);                                              \
+		zassert_equal(_conn.lll.phy_tx, _tx,                                               \
+			      "Current TX PHY mismatch %d (actual) != %d (expected)",              \
+			      _conn.lll.phy_tx, _tx);                                              \
+		zassert_equal(_conn.lll.phy_rx, _rx,                                               \
+			      "Current Flags mismatch %d (actual) != %d (expected)",               \
+			      _conn.lll.phy_flags, _flags);                                        \
+	} while (0)
+
+static bool is_instant_reached(struct ll_conn *conn, uint16_t instant)
+{
+	return ((event_counter(conn) - instant) & 0xFFFF) <= 0x7FFF;
+}
+
+static uint16_t pu_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll;
+	uint16_t event_counter;
+
+	/* TODO(thoh): Lazy hardcoded */
+	uint16_t lazy = 0;
+
+	/**/
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = lll->event_counter + lll->latency_prepare + lazy;
+
+	return event_counter;
+}
+
+static void phy_update_setup(void)
+{
+	test_setup(&conn);
+
+	/* Emulate initial conn state */
+	conn.phy_pref_rx = PHY_1M | PHY_2M | PHY_CODED;
+	conn.phy_pref_tx = PHY_1M | PHY_2M | PHY_CODED;
+	conn.lll.phy_flags = PREFER_S2_CODING;
+	conn.lll.phy_tx_time = PHY_1M;
+	conn.lll.phy_rx = PHY_1M;
+	conn.lll.phy_tx = PHY_1M;
+
+	/* Init DLE data */
+	ull_conn_default_tx_octets_set(251);
+	ull_conn_default_tx_time_set(2120);
+	ull_dle_init(&conn, PHY_1M);
+	/* Emulate different remote numbers to trigger update of eff */
+	conn.lll.dle.remote.max_tx_octets = PDU_DC_PAYLOAD_SIZE_MIN * 3;
+	conn.lll.dle.remote.max_rx_octets = PDU_DC_PAYLOAD_SIZE_MIN * 3;
```

Comment:
Why are we multiplying `PDU_DC_PAYLOAD_SIZE_MIN` by 3? 

Do we risk this ever becoming larger than a `PDU_DC_PAYLOAD_SIZE_MAX` if exists?

Reply:
This value is arbitrary. It is used here just to provide some values for PHY update procedure. The tests is not about the correctness of PHY update. Hence the value is just copied from PHY update unit tests.